tests/integration/common/state.rs

```diff
@@ -1,49 +1,92 @@
-    pub erc20_address: ContractAddress,
+use crate::common::blockifier_contracts::{
+    get_deprecated_erc20_contract_class, get_deprecated_feature_contract_class, get_feature_contract_class,
+};
+
+/// A struct to store all test state that must be maintained between initial setup, blockifier
+/// execution, and SNOS re-execution.
+///
+/// Some of this maintains the deployed contracts in a way that makes test writing easy, and some
+/// of it maintains state that is required for execution.
+#[derive(Debug)]
+pub struct TestState {
+    /// All deployed cairo0 contracts. Currently expects exactly one deploydment per class. String
+    /// represents the contract's name (such as file or class name, but is really arbitrary).
+    pub cairo0_contracts: HashMap<String, DeprecatedContractDeployment>,
+    /// All deployed cairo1 contracts. Currently expects exactly one deploydment per class. String
+    /// represents the contract's name (such as file or class name, but is really arbitrary).
+    pub cairo1_contracts: HashMap<String, ContractDeployment>,
+    /// The ERC20 fee contract deployments
+    pub fee_contracts: FeeContracts,
+
+    /// State initially created for blockifier execution
+    pub blockifier_state: CachedState<DictStateReader>,
+    /// All cairo0 compiled classes
+    pub cairo0_compiled_classes: HashMap<ClassHash, DeprecatedCompiledClass>,
+    /// All cairo1 compiled classes
+    pub cairo1_compiled_classes: HashMap<ClassHash, CasmContractClass>,
+}
+
+/// Struct representing a deployed cairo1 class
+#[derive(Debug)]
+pub struct ContractDeployment {
+    pub class_hash: ClassHash,
+    pub address: ContractAddress,
+    pub class: CasmContractClass,
+}
+
+/// Struct representing a deployed cairo0 class
+#[derive(Debug)]
+pub struct DeprecatedContractDeployment {
+    pub class_hash: ClassHash,
+    pub address: ContractAddress,
+    pub class: DeprecatedCompiledClass,
+}
+
+/// ERC20 contract deployments for Eth and Strk tokens, as well as the compiled class. Note that
+/// this is always a cairo0 contract.
+#[derive(Debug)]
+pub struct FeeContracts {
+    pub erc20_contract: DeprecatedCompiledClass,
+    pub eth_fee_token_address: ContractAddress,
+    pub strk_fee_token_address: ContractAddress,
 }
 
+/// Fixture to create initial test state in which all test contracts are deployed.
 #[fixture]
-pub fn initial_state(block_context: BlockContext) -> InitialState {
-    let account_without_validations_cairo1 = FeatureContract::AccountWithoutValidations(CairoVersion::Cairo1);
-    let account_without_validations_cairo0 = FeatureContract::AccountWithoutValidations(CairoVersion::Cairo0);
-    let test_contract_cairo1 = FeatureContract::TestContract(CairoVersion::Cairo1);
-    let test_contract_cairo0 = FeatureContract::TestContract(CairoVersion::Cairo0);
-    let erc20_cairo0 = FeatureContract::ERC20;
-
-    let state = test_state(
+pub async fn initial_state(block_context: BlockContext) -> TestState {
```

Comment:
What do you think about having a Cairo 0 specific state? I think it would be better to isolate it so that if we have a bug in Cairo1-related code we can more easily find the issue. We can disregard the ERC20 contract being Cairo 0 for now (the Cairo1 test will have 1 Cairo0 contract so it's not fully independent, that's life) but at least the Cairo 0 test is 100% Cairo 0.